tests/unit/test_http_client.py

```diff
@@ -36,144 +34,44 @@
 #####
 # Tests for simple client instantiation
 #####
-args = 'scopes,provide_loop'
-params = [
-    [['not-a-real-scope'], True],
-    [['not-a-real-scope'], False],
-    [None, True],
-    [None, False],
-]
-
-
-@pytest.mark.parametrize(args, params)
-def test_http_client_default(scopes, provide_loop, event_loop, fake_keyfile,
-                             fake_keyfile_data, mock_credentials):
+@pytest.mark.parametrize('provide_session', [True, False])
+def test_http_client_default(provide_session, mocker):
     """AIOGoogleHTTPClient is created with expected attributes."""
-    loop = None
-    if provide_loop:
-        loop = event_loop
-
-    client = http_client.AIOGoogleHTTPClient(
-        keyfile=fake_keyfile, scopes=scopes, loop=loop
-    )
-    assert fake_keyfile_data == client._keydata
-
-    if not scopes:
-        scopes = ['cloud-platform']
-    exp_scopes = [f'https://www.googleapis.com/auth/{s}' for s in scopes]
-
-    assert exp_scopes == client.scopes
-
-    if not provide_loop:
-        loop = asyncio.get_event_loop()
-    assert loop == client._loop
-
-    assert isinstance(client._session, aiohttp.client.ClientSession)
-    assert not client.token
-    assert not client.expiry
-
-    client._session.close()
-
-
-def test_http_client_raises_json(tmpdir, caplog):
-    """Client initialization raises when keyfile not valid json."""
-    caplog.set_level(logging.DEBUG)
-
-    tmp_keyfile = tmpdir.mkdir('keys').join('broken_keyfile.json')
-    tmp_keyfile.write('broken json')
-
-    with pytest.raises(exceptions.GCPGordonJanitorError) as e:
-        http_client.AIOGoogleHTTPClient(keyfile=tmp_keyfile)
-
-    e.match(f'Keyfile {tmp_keyfile} is not valid JSON.')
-    assert 1 == len(caplog.records)
-
-
-def test_http_client_raises_not_found(tmpdir, caplog):
-    """Client initialization raises when keyfile not found."""
-    caplog.set_level(logging.DEBUG)
+    session = None
+    if provide_session:
+        session = aiohttp.ClientSession()
 
-    tmp_keydir = tmpdir.mkdir('keys')
-    no_keyfile = os.path.join(tmp_keydir, 'not-existent.json')
-
-    with pytest.raises(exceptions.GCPGordonJanitorError) as e:
-        http_client.AIOGoogleHTTPClient(keyfile=no_keyfile)
-
-    e.match(f'Keyfile {no_keyfile} was not found.')
-    assert 1 == len(caplog.records)
-
-
-#####
-# Tests & fixtures for access token handling
-#####
-@pytest.fixture
-def mock_parse_expiry(mocker, monkeypatch):
-    mock = mocker.MagicMock(oauth_client, autospec=True)
-    mock._parse_expiry.return_value = datetime.datetime(2018, 1, 1, 12, 0, 0)
-    monkeypatch.setattr('gordon_janitor_gcp.http_client._client', mock)
-    return mock
+    creds = mocker.Mock(auth.GoogleAuthClient, autospec=True)
+    creds._session = aiohttp.ClientSession()
+    client = http_client.AIOGoogleHTTPClient(creds=creds, session=session)
 
+    if provide_session:
+        assert id(session) == id(client._session)
```

Comment:
I think this could be written as `assert session is client._session`. I don't know which is more pythonic, this is a nitpick anyway.

Reply:
wat:

```
>           assert id(creds._session) is id(client._session)
E           AssertionError: assert 4556486640 is 4556486640
E            +  where 4556486640 = id(<aiohttp.client.ClientSession object at 0x10f9677f0>)
E            +    where <aiohttp.client.ClientSession object at 0x10f9677f0> = <Mock spec='GoogleAuthClient' id='4556486584'>._session
E            +  and   4556486640 = id(<aiohttp.client.ClientSession object at 0x10f9677f0>)
E            +    where <aiohttp.client.ClientSession object at 0x10f9677f0> = <gordon_janitor_gcp.http_client.AIOGoogleHTTPClient object at 0x10f9679b0>._session
```

Reply:
nvm I'm an idiot. I'll go get coffee now.